[Issue #154] feat: wallet creation front end

chedieck opened the pull request:
Description:
Front end to create wallets.

Test plan:
Create some paybuttons, maybe some with an ecash address, others with a bitcoin cash address, others with both. Then try to create a wallet that include some of these addresses. Note that:
- Errors, such as "name not provided" should appear as a red text in the bottom of the form, such as it is being done on the "paybutton create" form;
- There should only be possible to create a wallet and set it as default for NETWORK if it contains an NETWORK address (where NETWORK is ecash or bitcoin cash);
- Editing the HTML before submitting the form, in order to try to set it as default for NETWORK when there are no NETWORK addresses selected should give the error `Default NETWORK wallet must have some NETWORK address.`


Depends on:
- [x] #354

ScottMcDermid commented on components/Wallet/WalletForm.tsx:
```diff
@@ -1,54 +1,170 @@
+    if (selectedPaybuttonIdList === undefined) return false
+    for (const selectedPaybuttonId of selectedPaybuttonIdList) {
+      let paybutton = userPaybuttons.find((pb) => pb.id === Number(selectedPaybuttonId))
+      if (paybutton === undefined) {
```
Can we avoid using `continue`? We could add this `if` condition to the next line:
```js
if (paybutton !== undefined && paybutton.addresses.some... )
```

chedieck replied:
Addressed.

ScottMcDermid reviewed (changes requested):
Tested, and validation works as expected. Just one code comment.

Klakurka reviewed (changes requested):
- Let's fade out the text for checkbox items that aren't selectable. It's not obvious without first adding a new button that is selectable and seeing the contrast in the checkbox.
![image](https://user-images.githubusercontent.com/4713204/213840530-578c289f-7b70-4271-b14a-75fac2bdc61c.png)
- There's a weird dot in the middle of the dialog (see above).
- Remove 'exists for this user' from the duplicate name error message. The errors are for the people using this, not for us (the admins / devs).

chedieck commented:
>Let's fade out the text for checkbox items that aren't selectable. It's not obvious without first adding a new button that is selectable and seeing the contrast in the checkbox.

Addressed. @johnkuney please [check this commit](https://github.com/PayButton/paybutton-server/pull/356/commits/c7c4a0542d3a429598e1bba6e67b59a3f87a367a) to see if this is the right way to do it.


> There's a weird dot in the middle of the dialog (see above).

Fixed.

>  Remove 'exists for this user' from the duplicate name error message. The errors are for the people using this, not for us (the admins / devs).

Fixed.

johnkuney commented:
Looks great! I just tweaked the disabled style a tad and added a couple other style tweaks hope thats cool. 
- Added an explainer blurb at the top as its not clear why items may be disabled
- Added a line if there are no buttons yet, with a link to the buttons page
- Moved the "make default" checkboxes down on the edit form so it matches the create form

johnkuney commented:
hmmm guess I broke a check with those commits...not sure why

chedieck commented:
Thanks for the changes John. About the building breaking:
 
Let's leave on this PR the changes you made regarding only the **create** wallet logic, that is, exclude the last two commits ([2251315](https://github.com/PayButton/paybutton-server/pull/356/commits/2251315c4e7bd2b9e7d2e7f5abfe5b60a0166b80) and [484a298](https://github.com/PayButton/paybutton-server/pull/356/commits/484a298ee546ee7a32177808b59338320a7831c0)). Then, after this is approved, you can do it on [the next PR](https://github.com/PayButton/paybutton-server/pull/357), regarding editing a wallet's paybuttons, that has this as a requirement.

It's very likely that those changes being made here would lead to conflicts, and on this branch editing some wallets paybuttons does not work, so it's better to use a state where it works (next PR) and then making the changes to better the style.

Klakurka reviewed (approved):
Ok I'll approve but I made this task for the UI improvements that should go in soon: https://github.com/PayButton/paybutton-server/issues/371
